lte/gateway/c/session_manager/OperationalStatesHandler.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <folly/dynamic.h>
+#include <folly/json.h>
+#include <google/protobuf/util/json_util.h>
+
+#include "SessionStore.h"
+
+namespace magma {
+const std::string TYPE                  = "type";
+const std::string SUBSCRIBER_STATE_TYPE = "subscriber_state";
+const std::string DEVICE_ID             = "device_id";
+const std::string VALUE                 = "value";
+const std::string SESSION_ID            = "session_id";
+const std::string IMSI                  = "imsi";
+const std::string MSISDN                = "msisdn";
+const std::string IPV4                  = "ipv4";
+const std::string ACTIVE_POLICY_RULES   = "active_policy_rules";
+
+std::list<std::map<std::string, std::string>> get_operational_states(
+    magma::SessionStore* session_store) {
+  google::protobuf::util::JsonPrintOptions options;
+  options.add_whitespace = false;
+
+  std::list<std::map<std::string, std::string>> states;
+  auto session_map = session_store->read_all_sessions();
+  for (auto& it : session_map) {
+    const auto imsi = it.first;
+    std::map<std::string, std::string> state;
+    state[TYPE]                    = SUBSCRIBER_STATE_TYPE;
+    state[DEVICE_ID]               = imsi;
+    folly::dynamic sessions_by_apn = folly::dynamic::object;
+
+    for (auto& session : it.second) {
+      if (session->get_state() == magma::SESSION_ACTIVE) {
```

Comment:
Is there a reason we want to show active sessions alone, can we get the state till the session is actually removed from the system ? When the session is no longer active, is it immediately removed or removed after sometime ?

Reply:
@karthiksubraveti Currently the non-active sessions stay in the system for about 3-5 seconds. So that is why I figured we don't really have to report it.

Reply:
Ok, with some further thought, I decided to just put all sessions as for 5G we may have non-terminated sessions for a longer time in the future.

Reply:
Also it might help expose any issues by the time session moves from CREATED -> ACTIVE. When does this transition happen ?